source/common/common/shared_memory_hash_set.h

```diff
@@ -139,6 +139,23 @@ template <class Value> class SharedMemoryHashSet : public Logger::Loggable<Logge
                 num_values, control_->size);
       ret = false;
     }
+
+    uint32_t num_free_entries = 0;
+    uint32_t expected_free_entries = control_->options.capacity - control_->size;
+    for (uint32_t cell_index = control_->free_cell_index; cell_index != Sentinal;
+         cell_index = getCell(cell_index).next_cell) {
+      ++num_free_entries;
+      if (num_free_entries > expected_free_entries) {
+        // Don't infinite-loop with a corruption; break when we see there's a problem.
+        break;
+      }
+    }
+    if (num_free_entries != expected_free_entries) {
```

Comment:
Given that we don't have coverage for sanityCheck() itself, I would probably replace the logging and returning false with `RELEASE_ASSERT(num_free_entries == expected_free_entries);`. I would do the same elsewhere and then probably just have this be a void function. Sorry I didn't notice this before, but I think it would probably simplify the reading of this function and its use.

If you want to actually add tests for the corrupted cases, then this logic is more reasonable, but I kind of doubt you do. :) WDYT?

Reply:
sanityCheck() is checked with ASSERT_TRUE in the _test.cc.  Originally I had attach() and init() as separate entry-points and attach() would return false if sanityCheck() failed; now it throws, so I think the effect is the same, no?

Reply:
effectively yes, though, without actually having test coverage of the error paths, I'm not sure it's really worth all this extra logic vs. just hard asserting what you know to be true. It just makes the overall code simpler IMO. I don't feel very strongly about it either way.

Reply:
Ah, got it.  Actually I hit the last log with memset 0:

TEST_F(SharedMemoryHashSetTest, sanityCheckZeroedMemory) {
  setUp<TestValueZeroHash>();
  SharedMemoryHashSet<TestValueZeroHash> hash_set1(options_, true, memory_.get());
  memset(memory_.get(), 0, hash_set1.numBytes());
  ASSERT_FALSE(hash_set1.sanityCheck());
}

But the other ones are harder without poking around specific bytes.  I'll switch to RELEASE_ASSERT anyway, stay tuned.

Reply:
done